src/components/Settings/Settings.tsx

```diff
@@ -157,16 +157,20 @@ export class Settings extends Component<Props, State> {
     handleChooseDevice = (device: MediaDeviceInfo) => {
         const { onSettingsChanged, cameraOn, audioOn, screenShareOn } = this.props;
 
-        const camera =
-            device.kind === "videoinput" && cameraOn
-                ? device
-                : this.getCurrentDevice(ResourceTypes.VIDEO);
-        const audio =
+        const cameraDevice =
+            device.kind === "videoinput" ? device : this.getCurrentDevice(ResourceTypes.VIDEO);
+        const audioDevice =
             device.kind === "audioinput" && audioOn
                 ? device
                 : this.getCurrentDevice(ResourceTypes.AUDIO);
 
-        onSettingsChanged(true, false, screenShareOn, camera, audio);
+        onSettingsChanged(
+            true,
+            false,
+            screenShareOn,
+            cameraOn ? cameraDevice : undefined,
+            audioOn ? audioDevice : undefined
```

Comment:
its possible to search for a `cameraDevice` and send `undefined` to **onSettingsChanged** function if `cameraOn = false`.
you should change that logic.

Reply:
you are not able to choose device if camera is off. thats why I check device.kind

Reply:
but it changed by the recent commit